ports/servoshell/desktop/dialog.rs

```diff
@@ -45,55 +47,74 @@ impl Dialog {
                 .default_file_filter("All Supported Types");
         }
 
-        let dialog = FileDialog {
+        Dialog::File {
             dialog,
             multiple,
             response_sender,
-        };
+        }
+    }
 
-        Dialog::File(dialog)
+    pub fn new_alert_dialog(message: String, sender: IpcSender<()>) -> Self {
+        Dialog::Alert { message, sender }
     }
 
     pub fn update(&mut self, ctx: &egui::Context) -> bool {
         match self {
-            Dialog::File(dialog) => {
-                if dialog.dialog.state() == DialogState::Closed {
-                    if dialog.multiple {
-                        dialog.dialog.pick_multiple();
+            Dialog::File {
+                dialog,
+                multiple,
+                response_sender,
+            } => {
+                if dialog.state() == DialogState::Closed {
+                    if *multiple {
+                        dialog.pick_multiple();
                     } else {
-                        dialog.dialog.pick_file();
+                        dialog.pick_file();
                     }
                 }
 
-                let state = dialog.dialog.update(ctx).state();
+                let state = dialog.update(ctx).state();
 
                 match state {
                     DialogState::Open => true,
                     DialogState::Picked(path) => {
-                        if let Err(e) = dialog.response_sender.send(Some(vec![path])) {
+                        if let Err(e) = response_sender.send(Some(vec![path])) {
                             warn!("Failed to send file selection response: {}", e);
                         }
                         false
                     },
                     DialogState::PickedMultiple(paths) => {
-                        if let Err(e) = dialog.response_sender.send(Some(paths)) {
+                        if let Err(e) = response_sender.send(Some(paths)) {
                             warn!("Failed to send file selection response: {}", e);
                         }
                         false
                     },
                     DialogState::Cancelled => {
-                        if let Err(e) = dialog.response_sender.send(None) {
+                        if let Err(e) = response_sender.send(None) {
                             warn!("Failed to send cancellation response: {}", e);
                         }
                         false
                     },
                     DialogState::Closed => false,
                 }
             },
-        }
-    }
+            Dialog::Alert { message, sender } => {
+                let mut result = true;
```

Comment:
```suggestion
                let mut is_open = true;
```

We can call it `is_open` so it is clear what this represents.